Add to_out_variant tests

manuelcandales opened the pull request:
Differential Revision: D55816906

mergennachin reviewed (approved):
Land if the OSS CI passes

larryliu0820 commented on setup.py:
```diff
@@ -478,6 +478,7 @@ def run(self):
             cmake_args += [
                 "-DEXECUTORCH_BUILD_PYBIND=ON",
                 "-DEXECUTORCH_BUILD_KERNELS_QUANTIZED=ON",  # add quantized ops to pybindings.
+                "-DEXECUTORCH_BUILD_QUANTIZED_OPS_AOT = ON",
```
```suggestion
                "-DEXECUTORCH_BUILD_QUANTIZED_OPS_AOT=ON",
```

manuelcandales committed:
Add to_out_variant tests (#2880)

Summary: Pull Request resolved: https://github.com/pytorch/executorch/pull/2880

Reviewed By: mergennachin, larryliu0820

Differential Revision: D55816906
